src/global/useConfig.tsx

```diff
@@ -0,0 +1,18 @@
+import { createContext, useContext, useState } from 'react';
+import { useQuery } from 'react-query';
+
+const ConfigContext = createContext({});
+
+export const ConfigProvider = ({ children }: { children: any }) => {
```

Comment:
No, context providers in general have difficulty with SSR. 
So this is the purely client approach to getting server side values. 
To get any values we'd add them to the api route (which is server side)

That's where all this difficulty lies. client vs server aspects.

Reply:
hm so in a way, at risk of raining on the parade, this doesn't solve the problem for server routines that rely on env values, but I guess for those we can just call the "api" route directly... 🤔 messy, but could work

Reply:
not raining on the parade at all. this is exactly why I'm talking so much about this.

not 100% sure what "server routines" in this context is but in general it appears that:
server has access runtime process env
client side only has access to env vars from build time, because they all get inlined, including process.env